mantidproject/mantid
# Support sub dirs in System Tests

**Description of work.**

- Supports arbitrary subdirs in the system tests folder, so future developers can sort tests into folders as required
- Replace deprecated `imp` with `importlib`
- Cache OS env function, this was calling `lsb_release` for every test which was causing the system tests to take ~16s to get going locally. Removing this brings the time <1s.
- Does a coarse move of ISIS SANS files into sub-dirs

**To test:**
- Check that system tests still run (best to check on Windows esp.)
- Check that one of the moved files still ran on the CI job

*There is no associated issue.*




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
